crates/sui-framework/packages/move-stdlib/sources/vector.move

```diff
@@ -382,3 +382,91 @@ public macro fun zip_map_ref<$T1, $T2, $U>(
     zip_do_ref!($v1, $v2, |el1, el2| r.push_back($f(el1, el2)));
     r
 }
+
+/// Performs an in-place insertion sort on the vector `v` using the comparison function `le`.
+/// The sort is stable, meaning that equal elements will maintain their relative order.
+///
+/// Insertion sort is efficient for small vector, and can be faster than merge sort for almost
+/// sorted vectors (e.g. when the vector is already sorted or nearly sorted).
+public macro fun insertion_sort_by<$T>($v: &mut vector<$T>, $le: |&$T, &$T| -> bool) {
+    let v = $v;
+    let n = v.length();
+    if (n < 2) return ();
+    // do insertion sort
+    let mut i = 1;
+    while (i < n) {
+        let mut j = i;
+        while (j > 0 && $le(&v[j], &v[j - 1])) {
+            v.swap(j, j - 1);
+            j = j - 1;
+        };
+        i = i + 1;
+    }
+}
+
+/// Performs an in-place merge sort on the vector `v` using the comparison function `le`.
+/// Merge sort is efficient for large vectors, and is a stable sort.
+///
+/// Merge sort performs better than insertion sort for large vectors (~30 elements or more).
```

Comment:
Ah looks like we have a number of 30 here, maybe we should restate that above

Reply:
Will do!